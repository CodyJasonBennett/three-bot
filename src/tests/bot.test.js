import Bot from '../bot';

describe('bot', () => {
  it('loads events', () => {
    const events = Bot.prototype.loadEvents();
    expect([...events.keys()].length).not.toBe(0);
  });

  it('loads commands', () => {
    const commands = Bot.prototype.loadCommands();
    expect([...commands.keys()].length).not.toBe(0);
  });
});
