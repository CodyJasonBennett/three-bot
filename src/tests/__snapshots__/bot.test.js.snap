// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bot loads commands 1`] = `
Array [
  Object {
    "description": "Searches https://threejs.org/docs/ for docs matching query.",
    "execute": [Function],
    "name": "docs",
    "options": Array [
      Object {
        "description": "A query or class to search matching docs for",
        "name": "query",
        "required": true,
        "type": "string",
      },
    ],
  },
  Object {
    "description": "Searches https://threejs.org/examples/ for examples matching query.",
    "execute": [Function],
    "name": "examples",
    "options": Array [
      Object {
        "description": "Query to search matching examples for",
        "name": "query",
        "required": true,
        "type": "string",
      },
    ],
  },
  Object {
    "description": "Displays this bot's commands.",
    "execute": [Function],
    "name": "help",
  },
]
`;

exports[`bot loads events 1`] = `
Array [
  Object {
    "execute": [Function],
    "name": "message",
  },
  Object {
    "execute": [Function],
    "name": "raw",
  },
  Object {
    "execute": [Function],
    "name": "ready",
  },
]
`;
